src/ledctl.c

```diff
@@ -688,10 +714,14 @@ static ledctl_status_code_t _cmdline_ibpi_parse(int argc, char *argv[])
 static ledctl_status_code_t _cmdline_parse_non_root(int argc, char *argv[])
 {
 	int opt_index, opt = -1;
-	ledctl_status_code_t status = LEDCTL_STATUS_SUCCESS;
+	char *shortopts;
+	struct option *longopts;
+	optind = 1;
+
+	setup_options(&longopts, &shortopts, possible_params, ARRAY_SIZE(possible_params));
 
 	do {
-		opt = getopt_long(argc, argv, shortopt, longopt, &opt_index);
+		opt = getopt_long(argc, argv, shortopts, longopts, &opt_index);
```

Comment:
I think that we should check first parameter only. `getopt_long` should be executed only once.
We need to provide all params to prevent `getopt_long` from throwing errors